--- __init__.py.orig	2017-11-28 16:50:46.000000000 +0000
+++ __init__.py	2018-05-08 16:11:29.225765139 +0000
@@ -311,6 +311,18 @@
 ################################################################
 
 
+_ARCH_TRIPLET = {
+    'arm64': 'aarch64-linux-gnu',
+    'armhf': 'arm-linux-gnueabihf',
+    'i386': 'i386-linux-gnu',
+    'ppc64el': 'powerpc64le-linux-gnu',
+    'powerpc': 'powerpc-linux-gnu',
+    'amd64': 'x86_64-linux-gnu',
+    's390x': 's390x-linux-gnu',
+    'riscv64': 'riscv64-linux-gnu',
+}
+
+
 class CDLL(object):
     """An instance of this class represents a loaded dll/shared
     library, exporting functions using the standard C calling
@@ -344,7 +355,15 @@
         self._FuncPtr = _FuncPtr
 
         if handle is None:
-            self._handle = _dlopen(self._name, mode)
+            name = self._name
+            if name is not None and _os.getenv('SNAP_NAME', '') == 'snapcraft':
+                _name = _os.path.join(
+                    _os.getenv('SNAP'), 'usr', 'lib',
+                    _ARCH_TRIPLET.get(_os.getenv('SNAP_ARCH')),
+                    name)
+                if _os.path.exists(_name):
+                    name = _name
+            self._handle = _dlopen(name, mode)
         else:
             self._handle = handle
 
@@ -407,6 +426,7 @@
         _func_flags_ = _FUNCFLAG_STDCALL
         _func_restype_ = HRESULT
 
+
 class LibraryLoader(object):
     def __init__(self, dlltype):
         self._dlltype = dlltype
@@ -419,6 +439,13 @@
         return dll
 
     def __getitem__(self, name):
+        if _os.getenv('SNAP_NAME', '') == 'snapcraft':
+            _name = _os.path.join(
+                        _os.getenv('SNAP'), 'usr', 'lib',
+                        _ARCH_TRIPLET.get(_os.getenv('SNAP_ARCH')),
+                        name)
+            if _os.path.exists(_name):
+                name = _name
         return getattr(self, name)
 
     def LoadLibrary(self, name):
